src/containers/addNew.js

```diff
@@ -0,0 +1,193 @@
+          this.setState({
+            isDisabled: false,
+          });
+        } else
+          this.setState({
+            isDisabled: true,
+          });
+      },
+    );
+  };
+
+  validate = field => () => {
+    this.setState({
+      touched: { ...this.state.touched, [field]: true },
+    });
+  };
+
+  renderField = (label, field) => {
+    let classnames = null;
+
+    if (field === 'beerName')
+      classnames = classNames('form-group', {
+        'has-danger': this.state.touched.beerName && this.state.beerName === '',
+      });
+    else
+      classnames = classNames('form-group', {
+        'has-danger': this.state.touched.desc && this.state.desc === '',
+      });
+
+    return (
+      <div className={classnames}>
+        <label className="field-label">{label}</label>
+        <input
+          className="form-control"
+          type="text"
+          value={field === 'beerName' ? this.state.beerName : this.state.desc}
+          onChange={this.handleChange(field)}
+          onBlur={this.validate(field)}
+        />
+      </div>
+    );
+  };
+
+  renderNumberField = field => {
+    return (
+      <div className="form-group" id="rating-field">
+        <label className="field-label">Enter the rating</label>
+        <label className="stars">
+          <Ratings
+            rating={this.state.rating}
+            widgetEmptyColors="rgb(255,239,212)"
+            widgetHoverColors="rgb(173, 21, 21)"
+            changeRating={this.changeTheRating}
+            widgetDimensions='30px'
+          >
+            <Ratings.Widget />
+            <Ratings.Widget />
+            <Ratings.Widget />
+            <Ratings.Widget />
+            <Ratings.Widget />
+          </Ratings>
+        </label>
+        <input
+          type="number"
+          className="form-control rating-input"
+          onChange={this.handleChange(field)}
+          value={this.state.rating}
+          readOnly
+        />
+      </div>
+    );
+  };
+
+  renderImgUpload = field => {
+    return (
+      <div className="file-area">
+        <label className="field-label">Image: &nbsp; </label>
+        <div className="file-overlay">
+          <input
+            onChange={this.handleChange(field)}
+            className="file-input"
+            type="file"
+            accept=".png, .jpeg, .jpg"
+          />
+        </div>
+      </div>
+    );
+  };
+
+  onSubmit = e => {
+    e.preventDefault();
+
+    let temp = null,
+      file = null;
+
+    if (e.target[3].files && e.target[3].files[0]) {
+      temp = e.target[3].files[0].name;
+      file = e.target[3].files[0];
+    } else {
+      temp = null;
+      file = null;
+    }
+
+    let values = {
+      beerName: e.target[0].value,
+      rating: e.target[1].value,
+      desc: e.target[2].value,
+      img: temp,
+    };
+
+    this.props.addNewBeer(values, file).then(() => {
+      this.props.history.push('/beers');
+    });
+  };
```

Comment:
This function is totally unreadable. Why do you take values from event if you have them in state?

Reply:
I just didn't think to get them from state, I'll streamline it.